WalletWasabi/Services/SystemAwakeChecker.cs

```diff
@@ -66,51 +66,54 @@ private SystemAwakeChecker(CoinJoinManager coinJoinManager, Func<Task<IPowerSavi
 
 		protected override async Task ActionAsync(CancellationToken cancel)
 		{
-			IPowerSavingInhibitorTask? task = _powerSavingTask;
-
 			switch (CoinJoinManager.HighestCoinJoinClientState)
 			{
 				case CoinJoinClientState.Idle:
-					if (task is not null)
-					{
-						Logger.LogWarning("Computer idle state is allowed again.");
-						await task.StopAsync().ConfigureAwait(false);
-						_powerSavingTask = null;
-					}
+					await ReleaseAllPreventionAsync().ConfigureAwait(false);
```

Comment:
L15: I would consider renaming the class from `SystemAwakeChecker` to something more fitting. `PowerSavingGuard`? `PowerSavingInhibitor`?

rationale: We do not really "check" whether system is awake but we are actively making a computer not to go to sleep (we inhibit sleep).

Reply:
`SleepInhibitor`?

Reply:
Can be, yes.